subworkflows/nf-core/fastq_create_umi_consensus_fgbio/main.nf

```diff
@@ -0,0 +1,204 @@
+
+        // sets bwaindex to correct input
+        bwaindex    = fasta_meta ? bwa_index ? Channel.fromPath(bwa_index).collect().map{ it -> [[id:it[0].baseName], it] } : BWAMEM1_INDEX.out.index : []
+        // check what's created:
+        bwaindexcheck    = bwaindex.dump(tag: 'bwa_index_to_map')
+        // check reads channel
+        //readscheck = BAM2FASTQ_PRE.out.reads.dump(tag: 'reads_channel')
+        // appropriately tagged interleaved FASTQ reads are mapped to the reference
+        // the aligner should be set with the following parameters "-p -K 150000000 -Y"
+        // to be configured in ext.args of your config
+        BWAMEM1_MEM_PRE ( BAM2FASTQ_PRE.out.fastq, bwaindex, false )
+        ch_versions = ch_versions.mix(BWAMEM1_MEM_PRE.out.versions)
+        aligned_bam = BWAMEM1_MEM_PRE.out.bam
+    } else {
+
+        BWAMEM2_INDEX ( fasta_meta )
+        ch_versions = ch_versions.mix(BWAMEM2_INDEX.out.versions)
+
+        // sets bwaindex to correct input
+        bwaindex    = fasta_meta ? bwa_index ? Channel.fromPath(bwa_index).collect().map{ it -> [[id:it[0].baseName], it] } : BWAMEM2_INDEX.out.index : []
+        // check what's created:
+        bwaindexcheck    = bwaindex.dump(tag: 'bwaindex')
+        // appropriately tagged interleaved FASTQ reads are mapped to the reference
+        // the aligner should be set with the following parameters "-p -K 150000000 -Y"
+        // to be configured in ext.args of your config
+        BWAMEM2_MEM_PRE ( BAM2FASTQ_PRE.out.fastq, bwaindex, false )
+        ch_versions = ch_versions.mix(BWAMEM2_MEM_PRE.out.versions)
+        aligned_bam = BWAMEM2_MEM_PRE.out.bam
+    }
+
+    // in order to tag mates information in the BAM file
+    // FGBIO tool ZipperBams is used to merge info from mapped and unmapped BAM files
+    ZIPPERBAMS_PRE ( FASTQTOBAM.out.bam, aligned_bam, fasta, dict )
+    ch_versions = ch_versions.mix(ZIPPERBAMS_PRE.out.versions)
+
+    // if using duplex UMI paired strategy must be used and therefore
+    // only BAM files with all reads paired will be accepted
+    // to avoid groupReadsByUmi throwing an error we must filter the zipped BAM
+
+    groupready_bam = Channel.empty()
+
+    if (duplex) {
+        // filter params are -f 1 i.e. paired and defined
+        // in config file
+        // samtools view module also needs a tuple with index
+        // creating a dummy one
+        dummyIndexFile = file('./dummy.bai')
+        dummyIndexFile.text = 'dummy'
+        // mapping it into the dummy input
+        dummy_bam_bai = ZIPPERBAMS_PRE.out.bam.map{meta, bam -> [meta, bam, dummyIndexFile]}
+        check_dummy_bam_bai = dummy_bam_bai.dump(tag: 'dummy_index')
+        // then applying samtools view to filter only paired reads
+        BAMFILTER ( dummy_bam_bai, [], [] )
+        ch_versions = ch_versions.mix(BAMFILTER.out.versions)
+        groupready_bam = BAMFILTER.out.bam
+        // deleting dummy file
+        dummyIndexFile.delete()
```

Comment:
here I'd be looking for more elegant solutions.
the issue is:
- an empty channel is not good because the type needs to be file
- cannot index this bam file because it's a bit of a special one (results from zipperbams, that merges tags from a uBAM and a mapped one): samtools index throws a number of errors due to this. so a normal indexing step and a join of the channels doesn't work.
creating a dummy file is the best I could come up with after trying the two above. but probably there's a more elegant way to code this?